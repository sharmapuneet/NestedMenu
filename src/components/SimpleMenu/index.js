import React, { useState } from 'react'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import Collapse from '@material-ui/core/Collapse'
import ExpandLess from '@material-ui/icons/ExpandLess'
import ExpandMore from '@material-ui/icons/ExpandMore'
import Drawer from '@material-ui/core/Drawer'
import { makeStyles } from '@material-ui/core/styles'
import { Link } from 'react-router-dom'
import menuItems from '../../data'

const useStyles = makeStyles({
  list: {
    width: '33.33%',
  },
  links: {
    textDecoration:'none',
    color: '#000',
    width: '100%',
  },
  menuHeader: {
    paddingLeft: '30px',
  },
  wrapperInner: {
    padding: '5px',
  },
})

function SubMenu(children) {
  const classes = useStyles()
  const [menu, setMenu] = useState({})
  const [selected, setSelected] = useState(false)

  const handleClick = item => {
    setMenu(prevMenu => (
      {[item]: !prevMenu[item]}
    ))
  }

  const updateSelected = selectedIndex => {
    setSelected(selectedIndex)
  }

  return children.map(subOption => (
    !subOption.children ?
      <div key={subOption.name}>
        <ListItem 
          button 
          key={subOption.name}
          selected={selected === subOption.name}
          onClick={() => {
            handleClick(subOption.name)
            updateSelected(subOption.name)
          }}
        >
          <Link 
            to={subOption.url}
            className={classes.links}>
            <ListItemText 
              inset 
              primary={subOption.name} 
            />
          </Link>
        </ListItem>
      </div> :
      <div key={subOption.name}>
        <ListItem
          button 
          onClick={() => {
            handleClick(subOption.name)
            updateSelected(subOption.name)
          }}
          selected={selected === subOption.name}
        >
          <ListItemText
            inset 
            primary={subOption.name}
          />
          {menu[subOption.name] ? 
            <ExpandLess /> :
            <ExpandMore />
          }
        </ListItem>
        <Collapse 
          in={menu[subOption.name]}
          timeout="auto" 
          unmountOnExit
          classes={{ wrapperInner: classes.wrapperInner }}
        >
          {SubMenu(subOption.children)}
        </Collapse>
      </div>
  ))
}

const SimpleMenu = () => {
  const classes = useStyles()

  return (
    <div className={classes.list}>
      <Drawer 
        variant="persistent" 
        anchor="left"
        open
        classes={{ paper: classes.list }}>
        <div>
          <List>
            <ListItem 
              key="menuHeading"
              divider
              disableGutters
            >
              <ListItemText
              className={classes.menuHeader}
                inset
                primary="Nested Menu"
              />
            </ListItem>
            {SubMenu(menuItems.data)}
          </List>
        </div>
      </Drawer>
    </div>
  )
}

export default SimpleMenu
