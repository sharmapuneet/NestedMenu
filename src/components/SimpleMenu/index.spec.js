import React from 'react';
import { shallow, render } from 'enzyme';
import SimpleMenu from './index';

describe('SimpleMenu', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SimpleMenu />)
    expect(wrapper).toMatchSnapshot();
  });

  it('should have two textfields', () => {
    const wrapper = shallow(<SimpleMenu />)
    expect(wrapper.find('Link')).toHaveLength(14);
  });
});