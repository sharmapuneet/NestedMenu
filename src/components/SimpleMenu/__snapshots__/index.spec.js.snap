// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleMenu should render correctly 1`] = `
<div
  className="makeStyles-list-1"
>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    classes={
      Object {
        "paper": "makeStyles-list-1",
      }
    }
    open={true}
    variant="persistent"
  >
    <div>
      <WithStyles(ForwardRef(List))>
        <WithStyles(ForwardRef(ListItem))
          disableGutters={true}
          divider={true}
          key="menuHeading"
        >
          <WithStyles(ForwardRef(ListItemText))
            className="makeStyles-menuHeader-3"
            inset={true}
            primary="Nested Menu"
          />
        </WithStyles(ForwardRef(ListItem))>
        <div
          key="Item1"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
            key="Item1"
          >
            <Link
              className="makeStyles-links-2"
              to="/item"
            >
              <WithStyles(ForwardRef(ListItemText))
                inset={true}
                primary="Item1"
              />
            </Link>
          </WithStyles(ForwardRef(ListItem))>
        </div>
        <div
          key="Item2"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
            key="Item2"
          >
            <Link
              className="makeStyles-links-2"
              to="/item"
            >
              <WithStyles(ForwardRef(ListItemText))
                inset={true}
                primary="Item2"
              />
            </Link>
          </WithStyles(ForwardRef(ListItem))>
        </div>
        <div
          key="Item3"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(ListItemText))
              inset={true}
              primary="Item3"
            />
            <ExpandMoreIcon />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(Collapse))
            timeout="auto"
            unmountOnExit={true}
          >
            <div
              key="Item3 - Child1"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                key="Item3 - Child1"
              >
                <Link
                  className="makeStyles-links-2"
                  to="/form"
                >
                  <WithStyles(ForwardRef(ListItemText))
                    inset={true}
                    primary="Item3 - Child1"
                  />
                </Link>
              </WithStyles(ForwardRef(ListItem))>
            </div>
            <div
              key="Item3 - Child2"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                key="Item3 - Child2"
              >
                <Link
                  className="makeStyles-links-2"
                  to="/item"
                >
                  <WithStyles(ForwardRef(ListItemText))
                    inset={true}
                    primary="Item3 - Child2"
                  />
                </Link>
              </WithStyles(ForwardRef(ListItem))>
            </div>
            <div
              key="Item3 - Child3"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                key="Item3 - Child3"
              >
                <Link
                  className="makeStyles-links-2"
                  to="/form"
                >
                  <WithStyles(ForwardRef(ListItemText))
                    inset={true}
                    primary="Item3 - Child3"
                  />
                </Link>
              </WithStyles(ForwardRef(ListItem))>
            </div>
          </WithStyles(ForwardRef(Collapse))>
        </div>
        <div
          key="Item4"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(ListItemText))
              inset={true}
              primary="Item4"
            />
            <ExpandMoreIcon />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(Collapse))
            timeout="auto"
            unmountOnExit={true}
          >
            <div
              key="Item4 - Child1"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                key="Item4 - Child1"
              >
                <Link
                  className="makeStyles-links-2"
                  to="/form"
                >
                  <WithStyles(ForwardRef(ListItemText))
                    inset={true}
                    primary="Item4 - Child1"
                  />
                </Link>
              </WithStyles(ForwardRef(ListItem))>
            </div>
            <div
              key="Item4 - Child2"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                key="Item4 - Child2"
              >
                <Link
                  className="makeStyles-links-2"
                  to="/item"
                >
                  <WithStyles(ForwardRef(ListItemText))
                    inset={true}
                    primary="Item4 - Child2"
                  />
                </Link>
              </WithStyles(ForwardRef(ListItem))>
            </div>
            <div
              key="Item4 - Child2 - SubChild1"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ListItemText))
                  inset={true}
                  primary="Item4 - Child2 - SubChild1"
                />
                <ExpandMoreIcon />
              </WithStyles(ForwardRef(ListItem))>
              <WithStyles(ForwardRef(Collapse))
                timeout="auto"
                unmountOnExit={true}
              >
                <div
                  key="Item4 - Child2 - SubSubChild1"
                >
                  <WithStyles(ForwardRef(ListItem))
                    button={true}
                    key="Item4 - Child2 - SubSubChild1"
                  >
                    <Link
                      className="makeStyles-links-2"
                      to="/form"
                    >
                      <WithStyles(ForwardRef(ListItemText))
                        inset={true}
                        primary="Item4 - Child2 - SubSubChild1"
                      />
                    </Link>
                  </WithStyles(ForwardRef(ListItem))>
                </div>
                <div
                  key="Item4 - Child2 - SubSubChild2"
                >
                  <WithStyles(ForwardRef(ListItem))
                    button={true}
                    key="Item4 - Child2 - SubSubChild2"
                  >
                    <Link
                      className="makeStyles-links-2"
                      to="/item"
                    >
                      <WithStyles(ForwardRef(ListItemText))
                        inset={true}
                        primary="Item4 - Child2 - SubSubChild2"
                      />
                    </Link>
                  </WithStyles(ForwardRef(ListItem))>
                </div>
                <div
                  key="Item4 - Child2 - SubSubChild3"
                >
                  <WithStyles(ForwardRef(ListItem))
                    button={true}
                    onClick={[Function]}
                  >
                    <WithStyles(ForwardRef(ListItemText))
                      inset={true}
                      primary="Item4 - Child2 - SubSubChild3"
                    />
                    <ExpandMoreIcon />
                  </WithStyles(ForwardRef(ListItem))>
                  <WithStyles(ForwardRef(Collapse))
                    timeout="auto"
                    unmountOnExit={true}
                  >
                    <div
                      key="Item4 - Child2 - SubSubChild3 - 1"
                    >
                      <WithStyles(ForwardRef(ListItem))
                        button={true}
                        key="Item4 - Child2 - SubSubChild3 - 1"
                      >
                        <Link
                          className="makeStyles-links-2"
                          to="/form"
                        >
                          <WithStyles(ForwardRef(ListItemText))
                            inset={true}
                            primary="Item4 - Child2 - SubSubChild3 - 1"
                          />
                        </Link>
                      </WithStyles(ForwardRef(ListItem))>
                    </div>
                  </WithStyles(ForwardRef(Collapse))>
                </div>
              </WithStyles(ForwardRef(Collapse))>
            </div>
          </WithStyles(ForwardRef(Collapse))>
        </div>
        <div
          key="Item5"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(ListItemText))
              inset={true}
              primary="Item5"
            />
            <ExpandMoreIcon />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(Collapse))
            timeout="auto"
            unmountOnExit={true}
          >
            <div
              key="Item5 - Child1"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                key="Item5 - Child1"
              >
                <Link
                  className="makeStyles-links-2"
                  to="/form"
                >
                  <WithStyles(ForwardRef(ListItemText))
                    inset={true}
                    primary="Item5 - Child1"
                  />
                </Link>
              </WithStyles(ForwardRef(ListItem))>
            </div>
            <div
              key="Item5 - Child2"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                key="Item5 - Child2"
              >
                <Link
                  className="makeStyles-links-2"
                  to="/item"
                >
                  <WithStyles(ForwardRef(ListItemText))
                    inset={true}
                    primary="Item5 - Child2"
                  />
                </Link>
              </WithStyles(ForwardRef(ListItem))>
            </div>
          </WithStyles(ForwardRef(Collapse))>
        </div>
        <div
          key="Item6"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(ListItemText))
              inset={true}
              primary="Item6"
            />
            <ExpandMoreIcon />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(Collapse))
            timeout="auto"
            unmountOnExit={true}
          >
            <div
              key="Item6 - Child1"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                key="Item6 - Child1"
              >
                <Link
                  className="makeStyles-links-2"
                  to="/form"
                >
                  <WithStyles(ForwardRef(ListItemText))
                    inset={true}
                    primary="Item6 - Child1"
                  />
                </Link>
              </WithStyles(ForwardRef(ListItem))>
            </div>
            <div
              key="Item6 - Child2"
            >
              <WithStyles(ForwardRef(ListItem))
                button={true}
                key="Item6 - Child2"
              >
                <Link
                  className="makeStyles-links-2"
                  to="/form"
                >
                  <WithStyles(ForwardRef(ListItemText))
                    inset={true}
                    primary="Item6 - Child2"
                  />
                </Link>
              </WithStyles(ForwardRef(ListItem))>
            </div>
          </WithStyles(ForwardRef(Collapse))>
        </div>
      </WithStyles(ForwardRef(List))>
    </div>
  </WithStyles(ForwardRef(Drawer))>
</div>
`;
